### PR TITLE
BIC-5476 adding Recurring Payment Get endpoint details

### PR DESCRIPTION
**Pull request by Shirish-Bambora**

### CODE REVIEW

File: data/na/swagger/1-0-5.yaml [hunk 1]
```diff
@@ -786,6 +786,34 @@ paths:
           $ref: '#/responses/MethodNotAllowed'
         500:
           $ref: '#/responses/InternalServerError'
+  /recurringPayment:
+  '/recurring/{accountId}':
+    get:
+      summary: Get account
+      description: 'Get a Recurring Payment Profile using account ID, also known as ID.'
```
Comment (GolnazIrannejad-Bambora) [inline data/na/swagger/1-0-5.yaml:793]: is it known as ID or Billing ID (from UI)?

Comment (Shirish-Bambora) [reply data/na/swagger/1-0-5.yaml:793]: it is ID
image

File: data/na/swagger/1-0-5.yaml [hunk 2]
```diff
@@ -2349,6 +2393,142 @@ definitions:
...
+  RecurringPaymentCustom:
+    properties:
+      order_number:
+        type: string
+        maxLength: 30
+        description: 'Order number'
+      comments:
+        type: string
+        maxLength: 2000
+        description: alphanumeric
```
Comment (GolnazIrannejad-Bambora) [inline data/na/swagger/1-0-5.yaml]: should description be something like comment about profile account? or something like that

File: data/na/swagger/1-0-5.yaml [hunk 3]
```diff
@@ -2349,6 +2393,142 @@ definitions:
...
+      ref1:
+        type: string
+        maxLength: 256
```
Comment (GolnazIrannejad-Bambora) [inline data/na/swagger/1-0-5.yaml:2467]: maybe similar to payment profile we can have "description: custom reference field 1"

File: data/na/swagger/1-0-5.yaml (hunk 2 above)
Comment (Shirish-Bambora) [reply data/na/swagger/1-0-5.yaml]: I can update that.

File: data/na/swagger/1-0-5.yaml [hunk 4]
```diff
@@ -2370,6 +2550,18 @@ definitions:
           $ref: '#/definitions/Detail'
       validation:
         $ref: '#/definitions/CardValidation'
+  RecurringPaymentErrorResponse:
+    properties:
+      code:
+        type: integer
+        format: int32
```
Comment (GolnazIrannejad-Bambora) [inline data/na/swagger/1-0-5.yaml:2562]: Should we mention possible values and what they means

File: data/na/swagger/1-0-5.yaml [hunk 5]
```diff
@@ -2349,6 +2393,142 @@ definitions:
...
+        type: string
+        maxLength: 256
+      ref5:
+        type: string
+        maxLength: 256
+  RecurringPaymentSchedule:
+    properties:
+      period:
+        type: string
+        description: 'The billing period is the regular interval at which credit card transactions are charged'
```
Comment (GolnazIrannejad-Bambora) [inline data/na/swagger/1-0-5.yaml]: according to recurring payment API guide "the period is unit of measurement for the frequency of the payment: day, week, month or year. "
I think we should have example of are possible values

File: data/na/swagger/1-0-5.yaml (hunk 4 above)
Comment (Shirish-Bambora) [reply data/na/swagger/1-0-5.yaml:2562]: I followed the same stuff as payments, it just shows the fields and its types, there are no examples.

Comment (GolnazIrannejad-Bambora) [reply data/na/swagger/1-0-5.yaml:2562]: yeah, I still think it's good information but I can't check the ERD right now to see if those data are in database or just hardcoded stuff. I'm ok if we decide not to add them.

File: data/na/swagger/1-0-5.yaml [hunk 6]
```diff
@@ -786,6 +786,34 @@ paths:
...
+      description: 'Get a Recurring Payment Profile using account ID, also known as ID.'
+      tags:
+        - Recurring Payment Account
+      operationId: getAccount
+      parameters:
+        - name: accountId
+          in: path
+          description: The account id. (aka ID)
+          required: true
+          type: string
```
Comment (LindenMikus) [inline data/na/swagger/1-0-5.yaml]: this should be int type

File: data/na/swagger/1-0-5.yaml [hunk 7]
```diff
@@ -2349,6 +2393,147 @@ definitions:
         description: the ID of the payment profile. You need to keep track of this value.
       card:
         $ref: '#/definitions/ProfileCards'
+  RecurringPaymentResponse:
```
Comment (LindenMikus) [inline data/na/swagger/1-0-5.yaml:2396]: I think we can just call it Payment type? I don't think the documentation needs to know its a DTO. similar for the other types

File: data/na/swagger/1-0-5.yaml [hunk 8]
```diff
@@ -828,6 +856,22 @@ responses:
     description: Internal Server Error
     schema:
       $ref: '#/definitions/ErrorResponse'
+  RecurringPaymentBadRequest:
```
Comment (LindenMikus) [inline data/na/swagger/1-0-5.yaml:859]: do we need these prefixed with 'Recurring'?

**Review by GolnazIrannejad-Bambora — APPROVED** [review]

File: data/na/swagger/1-0-5.yaml [hunk 9]
```diff
@@ -2349,6 +2393,147 @@ definitions:
...
+        format: integer
+        description: 'Unique account ID'
+      state:
+        type: string
+        maxLength: 6
+        description: 'The state of the Recurring Payment account'
+      language:
+        type: string
+        maxLength: 3
+        description: 'Account's billing language'
```
Comment (LindenMikus) [inline data/na/swagger/1-0-5.yaml:2409]: are we sure there is no issue with having a ' inside the single quotes?

File: data/na/swagger/1-0-5.yaml [hunk 10]
```diff
@@ -2349,6 +2393,147 @@ definitions:
...
+  RecurringPaymentSchedule:
+    properties:
+      period:
+        type: string
+        description: 'The period is unit of measurement for the frequency of the payment: day, week, month or year.'
+      increment:
+        type: integer
+        description: 'The frequency of the transactions'
+      start_date:
+        type: string
```
Comment (LindenMikus) [inline data/na/swagger/1-0-5.yaml:2494]: shouldnt type be date?

File: data/na/swagger/1-0-5.yaml [hunk 11]
```diff
@@ -2349,6 +2393,147 @@ definitions:
...
+        description: 'Card will be charged at the end of the month'
+      upcoming_transactions:
+        type: array
+        maxLength: 3
+        items:
+          $ref: '#/definitions/UpcomingTransactionRecord'
+  UpcomingTransactionRecord:
+    properties:
+      billing_date:
+        type: string
```
Comment (LindenMikus) [inline data/na/swagger/1-0-5.yaml:2516]: type date?

File: data/na/swagger/1-0-5.yaml [hunk 12]
```diff
@@ -2349,6 +2393,147 @@ definitions:
...
+        description: 'Start date of the billing cycle'
+      billing_end:
+        type: string
```
Comment (LindenMikus) [inline data/na/swagger/1-0-5.yaml:2519]: type date?

File: data/na/swagger/1-0-5.yaml [hunk 13]
```diff
@@ -2349,6 +2393,147 @@ definitions:
...
+        description: 'Last date of the billing cycle'
+      amount:
+        type: number
+        format: double
```
Comment (LindenMikus) [inline data/na/swagger/1-0-5.yaml:2523]: is there a format decimal for this and the others? double != decimal

File: data/na/swagger/1-0-5.yaml [hunk 14]
```diff
@@ -2370,6 +2555,18 @@ definitions:
           $ref: '#/definitions/Detail'
       validation:
         $ref: '#/definitions/CardValidation'
+  RecurringPaymentErrorResponse:
+    properties:
+      code:
+        type: integer
+        format: int32
```
Comment (LindenMikus) [inline data/na/swagger/1-0-5.yaml:2562]: why do we say int32 here but integer elsewhere? should be consistent

**Review by LindenMikus — CHANGES_REQUESTED** [review]

File: data/na/swagger/1-0-5.yaml (hunk 8 above)
Comment (Shirish-Bambora) [reply data/na/swagger/1-0-5.yaml:859]: I feel like payment is a very generic term, also Recurring Payment is the product offering, thats why I am putting it as prefix, also BadRequest is already used for existing definition.

File: data/na/swagger/1-0-5.yaml (hunk 9 above)
Comment (Shirish-Bambora) [reply data/na/swagger/1-0-5.yaml:2409]: I am not sure, I will check the yaml how it is handled, else I can merge and test it on FS branch, and then try to fix it.

File: data/na/swagger/1-0-5.yaml (hunk 10 above)
Comment (Shirish-Bambora) [reply data/na/swagger/1-0-5.yaml:2494]: no, we have datetime response. that would be a string.
image

File: data/na/swagger/1-0-5.yaml (hunk 12 above)
Comment (Shirish-Bambora) [reply data/na/swagger/1-0-5.yaml:2519]: same as above.

File: data/na/swagger/1-0-5.yaml (hunk 13 above)
Comment (Shirish-Bambora) [reply data/na/swagger/1-0-5.yaml:2523]: this amount as a property is used in other places, and thats how it is defined, as I can not test locally, I am just copying this.
image

File: data/na/swagger/1-0-5.yaml (hunk 14 above)
Comment (Shirish-Bambora) [reply data/na/swagger/1-0-5.yaml:2562]: not sure, copied it from elsewhere.

**Review by LindenMikus — APPROVED** [review]

**Pull request merged**